File: urbanairship-sdk/src/test/java/com/urbanairship/analytics/InstallAttributionEventTest.java
/* Copyright 2017 Urban Airship and Contributors */

package com.urbanairship.analytics;

import com.urbanairship.BaseTestCase;

import org.json.JSONException;
import org.junit.Test;

import static junit.framework.Assert.assertEquals;

public class InstallAttributionEventTest extends BaseTestCase {

    @Test
    public void testData() throws JSONException {
        InstallAttributionEvent event = new InstallAttributionEvent("referrer");
        EventTestUtils.validateEventValue(event, "google_play_referrer", "referrer");
    }

    @Test
    public void testType() {
        InstallAttributionEvent event = new InstallAttributionEvent("referrer");
        assertEquals("install_attribution", event.getType());
    }
}

File: urbanairship-core/src/test/java/com/urbanairship/analytics/InstallAttributionEventTest.kt
/* Copyright Airship and Contributors */
package com.urbanairship.analytics

import com.urbanairship.BaseTestCase
import com.urbanairship.analytics.EventTestUtils.validateEventValue
import org.junit.Assert.assertEquals
import org.junit.Test

public class InstallAttributionEventTest public constructor() : BaseTestCase() {

    @Test
    public fun testData() {
        val event = InstallAttributionEvent("referrer")
        validateEventValue(event, "google_play_referrer", "referrer")
    }

    @Test
    public fun testType() {
        val event = InstallAttributionEvent("referrer")
        assertEquals("install_attribution", event.type.reportingName)
    }
}
